### DISCUSSION TITLE
Is it possible to measure the time to receive a callback?

### DISCUSSION THREAD

**Discussion by ValMati** [artilleryio/artillery#1943] [Artillery]
Hello,

we are considering using Artillery to load test one of our backend applications. The truth is that Artillery looks very good and we like a lot what we are reading. We just have a question about how to test one of our scenarios.

Some of the tasks that our application does take a lot of time (between 5-10 minutes). The client sends a request, an identifier is returned, the task starts processing and when the task is finished the service does a callback to send the result.

![image](https://github.com/artilleryio/artillery/assets/18322719/ea657009-83f1-4fa3-a363-587965962ee4)

Can Artillery test the time between sending the request and receiving the response?

Thank you very much

**Comment by hassy** [comment]
hi @ValMati 👋 Is this for an HTTP service? or something async like WebSocket/Socket.IO-based?

**Reply by ValMati** [reply]
Both the initial call and the callback are HTTP requests. I know this is not the best option, but it was done this way because of the limitations of one of the clients.

**Reply by hassy** [reply]
Ah okay, so the service uses a webhook to notify the client of a job's completion. This is a fairly common scenario, but we don't yet support it out of the box in Artillery. I can think of a couple of ways to do it with a bit of custom code, e.g. if you have control over the server that receives the webhook, you could do this:

1. Artillery scenario sends a request to the service, gets an ID back. 
2. Run a [`function`](https://www.artillery.io/docs/reference/engines/http#function-actions-and-beforescenario--afterscenario-hooks) hook. Inside this hook:
    * Store the timestamp + ID
    * Poll the service that receives webhooks on an endpoint that receives an ID and returns the timestamp of when the webhook for that ID was received (or 404)
    * When that timestamp is received, emit [a custom histogram metric](https://www.artillery.io/docs/reference/extension-apis#tracking-custom-metrics) of the delta between two timestamps. That will show the distribution of times it took to process these jobs.

**Reply by ValMati** [reply]
Thanks for the answer. I'll try to implement it, although I've just started with Artillery and I don't even control the simplest scenarios yet.

Thank you very much!
